protocol/value_test.go

```diff
@@ -76,3 +81,166 @@ func vmap(kt, vt wire.Type, items ...wire.MapItem) wire.Value {
 func vitem(k, v wire.Value) wire.MapItem {
 	return wire.MapItem{Key: k, Value: v}
 }
+
+// testValue wraps around wire.Value to allow for stream encoding/decoding
+// testing.
+type testValue struct {
+	val wire.Value
+}
+
+func testvalue(val wire.Value) testValue {
+	return testValue{val: val}
+}
+
+func (v testValue) streamEncodeStruct(sw *binary.StreamWriter) error {
```

Comment:
I couldn't add these stream methods directly onto the `wire.Value` object because it would have caused an import cycle. Wrapped it here and made it explicit that this is only for testing since objects using the wire.Value intermediate representation should continue to use the binary.Encode and binary.Decode methods.